### ISSUE TITLE
pytube.exceptions.RegexMatchError issue

### ISSUE THREAD

**Issue by TNTprizz** [pytube/pytube#1059]
I am on the latest version of pytube which is 10.9.3
It shows this error:
![圖片](https://user-images.githubusercontent.com/62872282/126439624-9b3fbe4d-5eec-408b-a10d-45ba8f0ba7d5.png)

> Traceback (most recent call last):
>  File "/home/pi/discord/TOS-DOS/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
>    ret = await coro(*args, **kwargs)
>  File "/home/pi/discord/src/cogs/sigma.py", line 176, in play
>    vid = pytube.YouTube(url) # Get the video of the url
>  File "/home/pi/discord/TOS-DOS/lib/python3.8/site-packages/pytube/__main__.py", line 71, in __init__
>    self.video_id = extract.video_id(url)
>  File "/home/pi/discord/TOS-DOS/lib/python3.8/site-packages/pytube/extract.py", line 162, in video_id
>    return regex_search(r"(?:v=|\/)([0-9A-Za-z_-]{11}).*", url, group=1)
>  File "/home/pi/discord/TOS-DOS/lib/python3.8/site-packages/pytube/helpers.py", line 129, in regex_search
>    raise RegexMatchError(caller="regex_search", pattern=pattern)
> pytube.exceptions.RegexMatchError: regex_search: could not find match for (?:v=|\/)([0-9A-Za-z_-]{11}).*

That I cannot get the video object using the url.
It happens randomly.
And reinstallation is performed for a thousand times.
1. What is happening and any quick fixes?
2. Will it be fixed?

**Comment by TNTprizz** [comment]
Ah, I fixed the issue by editing the following line in extract.py(line 162):(NOT)

> return regex_search(r"(?:v=|/)([0-9A-Za-z_-]{11}).", url, group=1)

to

> return url.split("v=")[1]

The problem is the `pytube.YouTube` cannot retrieve video data, so it causes HTTP404ERROR and somehow raise `regexmatcherror`
but still hope the issue(or features) will be fixed!

**Comment by glubsy** [comment]
What was the URL that caused the issue in the first place?

**Comment by TNTprizz** [comment]
erm it just happens randomly
I think it means it cannot get the url of the video from a playlist because of some HTTP404error

**Comment by tfdahlin** [comment]
@TNTprizz I'm going to need the URL you're using to create the `YouTube` object to try and fix this. The RegexMatchError should only be generated if it can't find a valid videoId in the URL you're parsing. For example, `YouTube('https://example.com/')` will raise this exception because there is no videoId in the URL.

**Comment by TNTprizz** [comment]
![圖片](https://user-images.githubusercontent.com/62872282/126860311-afd83b4f-7734-447a-9196-5ec9193a6f5e.png)
The playlist: https://www.youtube.com/playlist?list=PLm7QJ627j_qXNHQ-9N_WVSwleWValfh7a

It wasn't the link, the problem is it happens randomly when I am getting information of the videos of the playlist

**Comment by tfdahlin** [comment]
This might be solved by the fix for the 404 error, so I'm going to wait on investigating this until that issue is resolved.

**Comment by tfdahlin** [comment]
I think this was resolved in the latest version of pytube (v11.0.0), as I can't replicate this. I'm going to close the issue. If you still have this problem, please open a new issue
